lib/hanami/slice.rb

```diff
@@ -99,17 +99,20 @@ def define_container
       container = Class.new(Dry::System::Container)
       container.use :env
 
-      container.config.name = name
-      container.config.inflector = application.configuration.inflector
-
-      if root && File.directory?(root)
-        container.config.root = root
-        container.config.bootable_dirs = ["config/boot"]
-
-        container.config.auto_register = ["lib/#{namespace_path}"]
-        container.config.default_namespace = namespace_path.tr("/", ".")
-
-        container.load_paths! "lib"
+      container.configure do |config|
+        config.name = name
+        config.inflector = application.configuration.inflector
+
+        if root && File.directory?(root)
+          config.root = root
+          config.bootable_dirs = ["config/boot"]
+
+          if File.directory?(root.join("lib"))
+            config.component_dirs.add "lib" do |dir|
+              dir.default_namespace = namespace_path.tr("/", ".")
```

Comment:
This line assumes UNIX like systems, should we use `::FIle::SEPARATOR` instead? This to guarantee cross platform compatibility.

Regarding the dot, do we have a central place where this namespace separator is defined and it can be referenced from here? This is both for maintenance perspective and also code clarity.

Reply:
Thanks @jodosha — I'm updating this now.

As for the dot, I've changed it to use `config.namespace_separator` (so it'll use the default value provided by dry-system), and added this card to make it possible for this config (and others) to be propagated to slices if the user configures something particular on the application container: https://trello.com/c/nQBwnUUX